lib/src/runner/configuration/load.dart

```diff
@@ -51,23 +51,50 @@ Configuration load(String path, {bool global: false}) {
 
   if (document is YamlMap) {
     var loader = new _ConfigurationLoader(document, source, global: global);
-    var config = loader.load();
-    var includePathNode = document.nodes["include"];
-    if (includePathNode != null && includePathNode.value is! String) {
+    return _processIncludeIfAny(loader.load(), document);
+  }
+  throw new SourceSpanFormatException(
+      "The configuration must be a YAML map.", document.span, source);
+}
+
+/// If an `include` node is contained in [node], merges and returns [config].
+Configuration _processIncludeIfAny(Configuration config, YamlMap node) {
+  var source = node.span.sourceUrl.toFilePath(windows: Platform.isWindows);
+  var includePathNode = node.nodes["include"];
+  if (includePathNode != null && includePathNode.value is! String) {
+    throw new SourceSpanFormatException(
+        "The 'include' field must be omitted or a String",
+        includePathNode.span);
+  }
+  if (includePathNode != null) {
+    Uri includeValue;
+    try {
+      includeValue = Uri.parse(includePathNode.value);
+    } on FormatException catch (_) {
       throw new SourceSpanFormatException(
-          "The 'include' field must be omitted or a String",
+          "Not a valid file URL: \"${includePathNode.value}\"",
           includePathNode.span,
           source);
     }
-    if (includePathNode != null) {
-      var basePath = p.join(p.dirname(path), includePathNode.value);
+    if (includeValue.isAbsolute) {
```

Comment:
I don't think we necessarily want to forbid absolute file URLs. I could imagine someone wanting to write `include: file:///usr/share/dart_test.yaml` in their global config, for example.

Reply:
Done.